cmd/lock-rest-client.go

```diff
@@ -89,17 +89,13 @@ func (client *lockRESTClient) Close() error {
 
 // restCall makes a call to the lock REST server.
 func (client *lockRESTClient) restCall(ctx context.Context, call string, args dsync.LockArgs) (reply bool, err error) {
-	values := url.Values{}
-	values.Set(lockRESTUID, args.UID)
-	values.Set(lockRESTOwner, args.Owner)
-	values.Set(lockRESTSource, args.Source)
-	values.Set(lockRESTQuorum, strconv.Itoa(args.Quorum))
-	var buffer bytes.Buffer
-	for _, resource := range args.Resources {
-		buffer.WriteString(resource)
-		buffer.WriteString("\n")
+	sort.Strings(args.Resources)
+	argsBytes, err := args.MarshalMsg(nil)
```

Comment:
This will be high traffic, so lets avoid the alloc if we can.
```suggestion
	argsBytes, err := args.MarshalMsg(metaDataPoolGet()[:0])
	defer metaDataPoolPut(argsBytes)
```